pkg/generate/template.go

```diff
@@ -150,7 +150,8 @@ type templateAPIVars struct {
 // code for a single top-level resource's API definition
 type templateCRDVars struct {
 	templateMetaVars
-	CRD *ackmodel.CRD
+	CRD                  *ackmodel.CRD
+	CompareIgnoredFields []string
```

Comment:
This is working fine, but i'm not sure if it's the best we can do here. Maybe i should hide this logic within the CRD object and use custom template functions to generate the compare code ? /cc @jaypipes

Reply:
Well, you could add a method to CRD like this:

https://github.com/aws/aws-controllers-k8s/blob/18f42be3be14b2d91363249d198dbca7616e45c2/pkg/model/crd.go#L275-L281

That way you wouldn't need to add any fields to the templateCRDVars.

Reply:
Aaaah, TIL - Didn't know functions with `func() bool` signature can be evaluated with the templating library.  Thank you!